### PR TITLE
SW-2992 Automatically create reports

### PR DESCRIPTION
**Pull request by sgrimm**
Once a day, check whether or not there are organizations with the `Reporter`
internal tag that don't yet have a report for the previous calendar quarter. If
there are any, create reports for them.

For ad-hoc testing in dev environments, advancing the test clock using the admin
UI will cause daily tasks to run.

This requires a minor change to the permission logic in `OrganizationStore`,
which previously didn't allow the system user to fetch organization details
because the system user doesn't support enumerating its organization and
facility roles.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #982** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/982" target="_blank">Graphite</a> 
    * **PR #983** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/983" target="_blank">Graphite</a> 
      * **PR #985** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/985" target="_blank">Graphite</a> 
        * **PR #986** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/986" target="_blank">Graphite</a> 
          * **PR #988** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/988" target="_blank">Graphite</a> 
            * **PR #989** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/989" target="_blank">Graphite</a> 
              * **PR #990** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/990" target="_blank">Graphite</a> 
                * **PR #991** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/991" target="_blank">Graphite</a>  👈
                  * **PR #992** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/992" target="_blank">Graphite</a> 
                    * **PR #994** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/994" target="_blank">Graphite</a> 
                      * **PR #995** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/995" target="_blank">Graphite</a> 
                        * **PR #996** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/996" target="_blank">Graphite</a> 
                          * **PR #997** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/997" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/991?utm_source=stack-comment).

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #982** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/982" target="_blank">Graphite</a> 
    * **PR #983** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/983" target="_blank">Graphite</a> 
      * **PR #985** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/985" target="_blank">Graphite</a> 
        * **PR #986** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/986" target="_blank">Graphite</a> 
          * **PR #988** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/988" target="_blank">Graphite</a> 
            * **PR #989** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/989" target="_blank">Graphite</a> 
              * **PR #990** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/990" target="_blank">Graphite</a> 
                * **PR #991** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/991" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/991?utm_source=stack-comment).

File: src/main/kotlin/com/terraformation/backend/customer/db/OrganizationStore.kt [hunk 1]
```diff
@@ -113,7 +126,7 @@ class OrganizationStore(
         .select(
             ORGANIZATIONS.asterisk(), facilitiesMultiset, internalTagsMultiset, totalUsersSubquery)
         .from(ORGANIZATIONS)
-        .where(listOfNotNull(ORGANIZATIONS.ID.`in`(organizationIds), condition))
+        .where(listOfNotNull(organizationIdCondition, condition))
```
Comment (charlie-tf) [inline src/main/kotlin/com/terraformation/backend/customer/db/OrganizationStore.kt:129]: Cool, this makes sense

**Review by charlie-tf — APPROVED** [review]
Looks great. Thank you

**Pull request merged**